### DISCUSSION TITLE
Is it possible to set the command-line flags inside a docker container?

### DISCUSSION THREAD

**Discussion by ghost** [prometheus/prometheus#11806] [Q&A] [answered]
In my current setting, I am not able to add command-line flags like `--storage.tsdb.retention.time` in a docker-compose.yaml file – don’t ask why. That’s why I need an alternative.

Is it possible to change it via the prometheus.yml config file?
Or is it possible to change the value with a .env-var like `STORAGE.TSDB.RETENTION.TIME`?

**Comment by JDTX0** [comment, marked as answer]
Configuring retention time via the YML config file isn't possible. See the following [discussion](https://github.com/prometheus/prometheus/issues/6188) for more info on that.

Environment variables are also not supported for configuring retention times, the only way is through the [flag](https://github.com/prometheus/prometheus/blob/main/cmd/prometheus/main.go#L318)

That being said, there are clever ways you can parse environment variables and pass them as flags with a custom entrypoint script. One such example can be found [here](https://github.com/hiveco/prometheus-docker)
